### PR TITLE
Add parameter check

### PR DESCRIPTION
**Pull request by howie6879**
Hi there:

There is a basic route looks like the following:

``` python
from sanic import Sanic
from sanic.response import text

app = Sanic()


@app.route("/")
async def test():
    return text('Hello World!')


if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8000)
```
It can be seen that despite the lack of the request parameter service can still run.

When you write a route, it is likely to forget to add a parameter called `request` in the function.

Maybe we can add a parameter check mechanism:

``` python
python  app.py

ValueError: Required parameter `request` missing in the test()
```

### CODE REVIEW

**Comment by bosskwei** [comment]
Well, I think this check is never needed, some ide will warn an unused warning if we do not use the request param next. So let the user decide whether to keep this param.

**Comment by howie6879** [comment]
Hi @bosskwei : 

Maybe you don't understand what I mean, as this example, although the service can be successfully started, but when I visit `http://0.0.0.0:8000/`, The program will throw a error:

``` python
TypeError: test() takes 0 positional arguments but 1 was given
```

So this parameter is necessary

**Comment by bosskwei** [comment]
@howie6879 My wrong, it's necessary..😅

**Comment by yunstanford** [comment]
It's pretty normal and conventional to have `request` as param. It's not very valuable to add extra overhead cost for checking this. And it might also break some libs built on top of that by breaking the convention.

**Comment by r0fls** [comment]
>It's not very valuable to add extra overhead cost for checking this.

Well, it's only happening during the route registration, so it wouldn't effect the actual response time. Might be worth it to add something like this to make the error more intuitive.

**Comment by yunstanford** [comment]
Ignore that part, :). Didn't look carefully. But still feel like it's not very good to break the convention. And I don't see a lot benefits. It's just my opinion. :). If you feel like it's useful, go ahead and get it merged

**Comment by howie6879** [comment]
@yunstanford @r0fls : 
As the routing increases,  it prevents the program from discovering errors while accepting the request

**Comment by yunstanford** [comment]
Sorry, I didn't get your point

**Comment by yunstanford** [comment]
Ah, sorry.. I think I misunderstood the PR and what you described. 

I was totally misunderstood what you described.. 

Sorry, the PR looks good.

**Comment by howie6879** [comment]
@yunstanford 

I apologize for my ambiguous description.

I mean it is better to check this parameter earlier......

And i need to learn English well... :100:

File: sanic/app.py [hunk 1]
```diff
@@ -143,12 +144,19 @@ def route(self, uri, methods=frozenset({'GET'}), host=None,
...
+                if stream:
+                    handler.is_stream = stream
+
+                self.router.add(uri=uri, methods=methods, handler=handler,
+                                host=host, strict_slashes=strict_slashes,
+                                version=version, name=name)
+                return handler
+            else:
+                raise ValueError(
+                    'Required parameter `request` missing in the {0}()'.format(
```
Comment (r0fls) [inline sanic/app.py]: Can you make this say ```Required parameter `request` missing in the {0}() route```? Otherwise LGTM

Comment (howie6879) [reply sanic/app.py]: Done

**Pull request merged**